llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/TaskExecutorService.java

```diff
@@ -213,6 +215,10 @@ public synchronized void setCapacity(int newNumExecutors, int newWaitQueueSize)
     numSlotsAvailable.addAndGet(newNumExecutors - maxParallelExecutors);
     maxParallelExecutors = newNumExecutors;
     waitQueue.setWaitQueueSize(newWaitQueueSize);
+    if (metrics != null) {
```

Comment:
I found only one case (test) where metrics could be passed in a null. So, I would wonder if it would make sense to mock metrics in that test and to always assume/verify that we have metrics being passed in?

Reply:
Yes, this makes sense. I think removing all those null checks just makes the code more readable. Done.